build-support/build-opts.js

```diff
@@ -0,0 +1,66 @@
+const execSync = require('sync-exec')
+const { env } = process
+const production = !!env.production
+const src = './src'
+const dest = env.dest ? env.dest : './dist'
+const noGlobals = !!env.noGlobals // don't generate globals bundle
+const noMap = !!env.noMap // don't generate .map files
+const codap = !!env.codap // include CODAP-specific modifications
+const assetsSrc = codap ? src + '/assets/img/*.*' : src + '/assets/**/*.*'
+const assetsDst = codap ? dest + '/img/' : dest
+
+const commit = execSync('git log -1 --pretty=format:"%H"').stdout
+const date = new Date()
+
+// CODAP builds append '.ignore' to js filenames to avoid CODAP compilation
+const codapOutputFileName = (webpackChunk) => {
+  return webpackChunk.chunk.name.match(/\.js$/) ? '[name].ignore': '[name]';
+}
+
+const replacementStrings = [
+  {
+    pattern: /__COMMIT__/g,
+    value: commit
+  },
+  {
+    pattern: /__DATE__/g,
+    value: date
+  }
+]
+
+const entry = codap
+  ? {
+    'js/app.js': './code/app.coffee',
+    'css/app': './style/app.styl'
+  }
+  : {
+    'js/app.js': './code/app.coffee',
+    'js/globals.js': './code/globals.coffee',
+    'css/app': './style/app.styl'
+  }
```

Comment:
Should use `noGlobals` and should be able to reduce the redundancy here. Something like:
```
const appEntries = {
  'js/app.js': './code/app.coffee',
  'css/app': './style/app.styl'
};
const entry = noGlobals
  ? appEntries
  : { 'js/globals.js': './code/globals.coffee', ...appEntries };
```